### PR TITLE
Add setQueryData to react-query-utils AND rename mutate to setQueryData

### PR DESCRIPTION
**Pull request by aericson**
Closes: #984 

### What are the changes and their implications?
1. Add `setQueryData` to `react-query-utils`.
     - Receives (resolver, params, newData, opts) as argument.
2. Some refactoring on `react-query-utils`.
3. Changed arguments to `getQueryCacheFunctions` to keep it more consistent with the other functions from this module and to reduce code duplication. This doesn't seem to be a documented function so I figured it would be ok to change it.
4. Renamed and `mutate` returned by `useQuery` to `setQueryData` as sugested [here](https://github.com/blitz-js/blitz/issues/984#issuecomment-706345381).

### Checklist
- [x] Tests added for changes
- [x] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes
    - https://github.com/blitz-js/blitzjs.com/pull/227

### CODE REVIEW

**Comment by flybayer** [comment]
Sweet, looks good to me!!

File: packages/core/src/utils/react-query-utils.ts [hunk 1]
```diff
@@ -114,6 +103,34 @@ export function invalidateQuery<TInput, TResult, T extends QueryFn>(
...
+
+  return new Promise((res) => {
+    queryCache.setQueryData(queryKey, newData)
+    let result: void | ReturnType<typeof queryCache.invalidateQueries>
+    if (opts.refetch) {
+      result = invalidateQuery(resolver, params)
+    }
+    if (isClient) {
+      // Fix for https://github.com/blitz-js/blitz/issues/1174
+      requestIdleCallback(() => {
```
Comment (jan-wilhelm) [inline packages/core/src/utils/react-query-utils.ts:125]: as shown in #1214 , requestIdleCallback is not supported in safari. Could you change this to use setTimeout maybe?

Comment (aericson) [reply packages/core/src/utils/react-query-utils.ts:125]: Thanks for taking a look at this!

I believe this was addressed by https://github.com/blitz-js/blitz/pull/1247 it's using the shim, not `window.requestIdleCallback`.

Comment (jan-wilhelm) [reply packages/core/src/utils/react-query-utils.ts:125]: yeah, exactly. That shim should work here too, I believe

Comment (aericson) [reply packages/core/src/utils/react-query-utils.ts:125]: Sorry, I'm a bit confused. It's already using that shim.

Comment (jan-wilhelm) [reply packages/core/src/utils/react-query-utils.ts:125]: oh, sorry. I didn't see the import and thought that this was implicitly using the function from "window". Thanks for clearing this up :)

Comment (aericson) [reply packages/core/src/utils/react-query-utils.ts:125]: No worries!

**Comment by aericson** [comment]
Added some new tests, will try to put the time into some docs tomorrow or the day after.

**Comment by aericson** [comment]
Docs added! From my side, this is ready to go. 😁

**Comment by flybayer** [comment]
Sweeet, thanks!!

**Pull request merged**